src/Http/DataTables/Tools/MoonsDataTable.php

```diff
@@ -160,7 +161,8 @@ public function html()
      */
     public function query()
     {
-        return MapDenormalize::moons()
+        return MapDenormalize::whereIn(
```

Comment:
Can you improve the query using a whereHas statement instead ?
This one will be bounced by engine if there are too many entries in the generated list.

Reply:
will sort it after merge, thanks for the contribution ;)